### PR TITLE
Cover reimplementation & bugfixing

### PR DESCRIPTION
**Pull request by Vliro**
Rewrites to stateful covers.

Added ItemHandler & Uncommented itemMerge, this made debugging life so much easier and fixed annoying gui glitches.

### CODE REVIEW

File: src/main/java/muramasa/antimatter/capability/IInteractHandler.java [hunk 1]
```diff
@@ -12,8 +12,8 @@
 
 @ParametersAreNonnullByDefault
 public interface IInteractHandler {
-
-    boolean onInteract(PlayerEntity player, Hand hand, Direction side, @Nullable AntimatterToolType type);
+    //ParsedSide represents the side as parsed by Utils.getInteractSide
+    boolean onInteract(PlayerEntity player, Hand hand, Direction side, @Nonnull Direction parsedSide, @Nullable AntimatterToolType type);
```
Comment (repo-alt) [inline src/main/java/muramasa/antimatter/capability/IInteractHandler.java:16]: Perhaps it should be called interactSide then?

File: src/main/java/muramasa/antimatter/cover/CoverInstance.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        return this.cover.getId();
+    }
+
+    public boolean isEmpty() {
+        return cover == Data.COVERNONE;
+    }
+
+    //Gets the backing cover.
+    //Because getCover().getCover() looks stupid
+    public Cover backing() {
```
Comment (repo-alt) [inline src/main/java/muramasa/antimatter/cover/CoverInstance.java:72]: backing is weird tho

File: src/main/java/muramasa/antimatter/capability/IInteractHandler.java (hunk 1 above)
Comment (Vliro) [reply src/main/java/muramasa/antimatter/capability/IInteractHandler.java:16]: I am not sure, so have two separate interacts?
I might actually not need ParsedSide though, it is used for wire cutter & wrench for connecting machines.

File: src/main/java/muramasa/antimatter/cover/CoverInstance.java (hunk 2 above)
Comment (Vliro) [reply src/main/java/muramasa/antimatter/cover/CoverInstance.java:72]: yeah but i always wrote getCover() to get the instance then getCover() to get the singleton, sounds stoopid

Comment (qubka) [reply src/main/java/muramasa/antimatter/cover/CoverInstance.java:72]: I am advising renaming to the asCover() because backing() even more confusing

Comment (Vliro) [reply src/main/java/muramasa/antimatter/cover/CoverInstance.java:72]: can change to asCover

Comment (qubka) [reply src/main/java/muramasa/antimatter/cover/CoverInstance.java:72]: Another idea it revert backing to getCover() and replace  
> public CoverInstance getCover(Direction side)
by the 
> public CoverInstance getCoverInstance(Direction side) or [getInstance]
which even should be better

File: src/main/java/muramasa/antimatter/capability/IInteractHandler.java (hunk 1 above)
Comment (Vliro) [reply src/main/java/muramasa/antimatter/capability/IInteractHandler.java:16]: for now i will keep it but as needed split it

**Pull request merged**